src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -18,57 +18,61 @@
 
 namespace Akka.Streams.Stage
 {
-    public interface IGraphStageWithMaterializedValue
+    public interface ILogicAndMaterializedValue<out TMaterialized>
```

Comment:
Variance can only be defined on interfaces. Should we make it a class to avoid boxing?